src/app/clusters/drlc-server/drlc-server.cpp

```diff
@@ -0,0 +1,554 @@
+/**
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "drlc-server.h"
+
+#include <app/AttributeAccessInterface.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/InteractionModelEngine.h>
+#include <app/MessageDef/StatusIB.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/CHIPError.h>
+
+namespace {
+// Spec-defined constraints
+constexpr uint8_t kMaxRandomStartMinutes    = 60;
+constexpr uint8_t kMaxRandomDurationMinutes = 60;
+constexpr uint8_t kMaxDefaulRandomStart     = 60;
+constexpr uint8_t kMaxDefaulRandomDuration  = 60;
+constexpr uint16_t kMaxTransitionDuration   = 1440;
+constexpr int16_t kMinTemperature           = -27315;
+constexpr int8_t kMinLoadAdjustment         = -100;
+constexpr int8_t kMaxLoadAdjustment         = 100;
+constexpr uint8_t kMaxPercentValue          = 100;
+} // anonymous namespace
+
+namespace chip {
+
+using Status = Protocols::InteractionModel::Status;
+
+namespace app {
+namespace Clusters {
+namespace DemandResponseLoadControl {
+
+using namespace Attributes;
+
+template <bool ClearLoadControlEventsRequestSupported>
+Instance<ClearLoadControlEventsRequestSupported>::Instance(EndpointId aEndpointId, Delegate & aDelegate) :
+    AttributeAccessInterface(MakeOptional(aEndpointId), Id), CommandHandlerInterface(MakeOptional(aEndpointId), Id),
+    mDelegate(aDelegate), mEndpoint(aEndpointId)
+{}
+
+template <bool ClearLoadControlEventsRequestSupported>
+Instance<ClearLoadControlEventsRequestSupported>::~Instance()
+{
+    // Just in case, make sure there are no dangling pointers to us.
+    Shutdown();
```

Comment:
`Shutdown` name feels like a regular command of the  DRLC cluster or like you would like to turn off the device.
in fact you just wish to deinit everything cleanly